tests/ui/UnreadIndicatorsTest.js

```diff
@@ -0,0 +1,521 @@
+        let renderedApp;
+        return signInAndGetAppWithUnreadChat()
+            .then((testInstance) => {
+                renderedApp = testInstance;
+
+                // Simulate a new report arriving via Pusher along with reportActions and personalDetails for the other participant
+                const NEW_REPORT_ID = 2;
+                const NEW_REPORT_CREATED_MOMENT = moment();
+                Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${NEW_REPORT_ID}`, {
+                    reportID: NEW_REPORT_ID,
+                    reportName: 'Chat Report',
+                    maxSequenceNumber: 1,
+                    lastReadSequenceNumber: 0,
+                    lastMessageTimestamp: NEW_REPORT_CREATED_MOMENT.utc(),
+                    lastMessageText: 'Comment 1',
+                    participants: [USER_C_EMAIL],
+                });
+                Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${NEW_REPORT_ID}`, {
+                    0: {
+                        actionName: CONST.REPORT.ACTIONS.TYPE.CREATED,
+                        automatic: false,
+                        sequenceNumber: 0,
+                        timestamp: NEW_REPORT_CREATED_MOMENT.unix(),
+                        reportActionID: NumberUtils.rand64(),
+                    },
+                    1: {
+                        actionName: CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT,
+                        actorEmail: USER_C_EMAIL,
+                        person: [{type: 'TEXT', style: 'strong', text: 'User C'}],
+                        sequenceNumber: 1,
+                        timestamp: NEW_REPORT_CREATED_MOMENT.add(5, 'seconds').unix(),
+                        message: [{type: 'COMMENT', html: 'Comment 1', text: 'Comment 1'}],
+                        reportActionID: NumberUtils.rand64(),
+                    },
+                });
+                Onyx.merge(ONYXKEYS.PERSONAL_DETAILS, {
+                    [USER_C_EMAIL]: TestHelper.buildPersonalDetails(USER_C_EMAIL, USER_C_ACCOUNT_ID, 'C'),
+                });
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                // Verify notification was created as the new message that has arrived is very recent
+                expect(LocalNotification.showCommentNotification.mock.calls.length).toBe(1);
+
+                // // Navigate back to the sidebar
+                return navigateToSidebar(renderedApp);
+            })
+            .then(() => {
+                // // Verify the new report option appears in the LHN
+                const optionRows = renderedApp.getAllByA11yHint('Navigates to a chat');
+                expect(optionRows.length).toBe(2);
+
+                // Verify the text for both chats are bold indicating that nothing has not yet been read
+                const displayNameTexts = renderedApp.queryAllByA11yLabel('Chat user display names');
+                expect(displayNameTexts.length).toBe(2);
```

Comment:
NAB: Here is a pretty cool assertion I came across:

```suggestion
                expect(displayNameTexts).toHaveLength(2);
```

I like it because it makes the intention of the assertion to be really clear.